libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -57,7 +59,8 @@ impl Greenlight {
             sdk_config,
             tls_config,
             signer,
-            scheduler: Mutex::new(None),
+            gl_client: Mutex::new(None),
+            node_client: Mutex::new(None),
```

Comment:
I wonder if we can get away with lazy initializing the connection. That would remove the need for the `Mutex`, since the grpc stubs are cloneable and will manage interior mutability (muxing requests onto the single connection).

I'm not yet familiar enough with the `tonic` API, but I'm sure we can use the `Endpoint` struct with its `connect_lazy` method.

Reply:
If its lazy-initialized, doesn't that mean a user action could incur an extra connection-opening time penalty?

Reply:
No, the connection initialization will have to be anyway. Also re-connection happens from time to time automatically.

Reply:
The schedule method connects to the scheduler and to the node gRPC endpoint.
I think if we remove the scheduler call (as we plan to) and use `connect_lazy` instead of `connect` here: 
https://github.com/Blockstream/greenlight/blob/main/libs/gl-client/src/node/mod.rs#L103

Then we might be able to do it.  But we also need that the new connect method will return the Client and not `Result` that might contain error.